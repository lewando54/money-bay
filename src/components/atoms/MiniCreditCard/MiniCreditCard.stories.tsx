/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import MiniCreditCard from './MiniCreditCard'

const meta = {
	component: MiniCreditCard,
	argTypes: {
		cardIssuer: { control: 'select', options: ['visa', 'mastercard'] },
        gradientColors: { control: 'array' }
	}
} satisfies Meta<typeof MiniCreditCard>

export default meta
type Story = StoryObj<typeof meta>

export const Visa: Story = {
	args: {
		cardIssuer: 'visa'
	}
}

export const Mastercard: Story = {
	args: {
		cardIssuer: 'mastercard',
	}
}

export const CustomGradientColors: Story = {
	args: {
		cardIssuer: 'visa',
        gradientColors: ['#f00', '#ff0']
	}
}