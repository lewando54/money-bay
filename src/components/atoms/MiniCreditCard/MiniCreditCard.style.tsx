import { StyleSheet } from 'react-native'
export default StyleSheet.create({
    container: {
        width: 72,
        height: 46,
        borderRadius: 5,
        padding: 10
    },
    cardIssuerLogo: {
        width: 25,
        height: 15,
        marginLeft: 'auto',
        marginTop: 'auto',
        resizeMode: 'contain'
    }
})