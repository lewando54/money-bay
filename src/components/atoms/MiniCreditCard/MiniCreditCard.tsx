import { Image, View } from 'react-native'
import miniCreditCardStyle from './MiniCreditCard.style'
import { LinearGradient } from 'expo-linear-gradient'

type TCardIssuer = 'visa' | 'mastercard'
type TGradientColors = Array<string>

interface IMiniCreditCardProps {
    cardIssuer?: TCardIssuer
    gradientColors?: TGradientColors
    testID?: string
}

export default function MiniCreditCard({
    cardIssuer='visa',
    gradientColors=['#C285AB', '#7AA1CE', '#1B1F6A'],
    testID
}: IMiniCreditCardProps){
    const cardIssuerLogo = 
    cardIssuer == 'visa' && require('./assets/visa.png') || 
    cardIssuer == 'mastercard' && require('./assets/mastercard.png')

    return (
        <LinearGradient start={{x: 0, y: 0}} end={{x: 1, y: 1}} colors={gradientColors} style={miniCreditCardStyle.container}>
            <Image source={cardIssuerLogo} style={miniCreditCardStyle.cardIssuerLogo}/>
        </LinearGradient>
    )
}
