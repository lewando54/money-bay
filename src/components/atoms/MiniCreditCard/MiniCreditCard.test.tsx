import React from 'react'
import renderer from 'react-test-renderer'

import MiniCreditCard from './MiniCreditCard'
describe('MiniCreditCard', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<MiniCreditCard />)
        expect(tree).toMatchSnapshot()
    })

    it('should render correct card issuer logo when chosen', () => {
        const tree = renderer.create(<MiniCreditCard cardIssuer='mastercard'/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render with custom gradient colors when provided', () => {
        const tree = renderer.create(<MiniCreditCard gradientColors={['#f00', '#ff0']}/>)
        expect(tree).toMatchSnapshot()
    })
})