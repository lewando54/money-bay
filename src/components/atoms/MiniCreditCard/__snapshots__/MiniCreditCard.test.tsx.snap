// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiniCreditCard should render correct card issuer logo when chosen 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4290938283,
        4286226894,
        4279967594,
      ],
      "endPoint": [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": [
        0,
        0,
      ],
    }
  }
  style={
    {
      "borderRadius": 5,
      "height": 46,
      "padding": 10,
      "width": 72,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 15,
        "marginLeft": "auto",
        "marginTop": "auto",
        "resizeMode": "contain",
        "width": 25,
      }
    }
  />
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`MiniCreditCard should render correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4290938283,
        4286226894,
        4279967594,
      ],
      "endPoint": [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": [
        0,
        0,
      ],
    }
  }
  style={
    {
      "borderRadius": 5,
      "height": 46,
      "padding": 10,
      "width": 72,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 15,
        "marginLeft": "auto",
        "marginTop": "auto",
        "resizeMode": "contain",
        "width": 25,
      }
    }
  />
</ViewManagerAdapter_ExpoLinearGradient>
`;

exports[`MiniCreditCard should render with custom gradient colors when provided 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    {
      "colors": [
        4294901760,
        4294967040,
      ],
      "endPoint": [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": [
        0,
        0,
      ],
    }
  }
  style={
    {
      "borderRadius": 5,
      "height": 46,
      "padding": 10,
      "width": 72,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 15,
        "marginLeft": "auto",
        "marginTop": "auto",
        "resizeMode": "contain",
        "width": 25,
      }
    }
  />
</ViewManagerAdapter_ExpoLinearGradient>
`;
